akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -910,8 +918,8 @@ private[akka] class ShardRegion(
       case Failure(_) => Success(Left(shardId))
     }
 
-  private def tryCompleteGracefulShutdown() =
-    if (gracefulShutdownInProgress && shards.isEmpty && shardBuffers.isEmpty) {
+  private def tryCompleteGracefulShutdown(): Unit =
+    if (shards.isEmpty && shardBuffers.isEmpty) {
```

Comment:
I think it's more robust that the `if (gracefulShutdownInProgress)` is included in `tryCompleteGracefulShutdown` since it's called from multiple places and is intended to be safe to be retried.

Same with `sendGracefulShutdownToCoordinator`, but we can rename the method to something like `trySendGracefulShutdownToCoordinator` or or we can rename both methods to `completeGracefulShutdownIfInProgress` and `sendGracefulShutdownToCoordinatorIfInProgress`.

Reply:
👍  for appending `IfInProgress`